add `TBReader` for convinient deserialization

Eggiverse opened the pull request:
This is another solution for #91.

It introduces a `TBReader` for convinient deserialization.

Eggiverse committed:
add `TBReader` for convinient deserialization

PhilipVinc commented on src/Deserialization/reader.jl:
```diff
@@ -0,0 +1,7 @@
+    logdir::String
+end
+
+TBReadable = Union{TBLogger, TBReader}
```
I’m not a fun of unions, as they are not extensible, but I think for now that is ok.,.

PhilipVinc reviewed (commented):
Can you add a test just to be sure it’s working?

Eggiverse committed:
Add test for `TBReader` and fix other tests

Event files are now properly closed in unit tests.
`TBEventFileIterator` now closes event file when iteration is finished.

PhilipVinc commented:
The failures are unrelated?

PhilipVinc commented:
hmm tests pass for me locally.
Maybe it was a post 1.6 release mess?

PhilipVinc commented:
I think this works. I will merge and try to fix tests

PhilipVinc commented:
Thanks a lot!
